data-prepper-core/src/test/java/com/amazon/dataprepper/meter/EMFLoggingMeterRegistryTest.java

```diff
@@ -0,0 +1,403 @@
+        registry.more().timer("functionTimer", Tags.empty(), 1d, Number::longValue, Number::doubleValue, TimeUnit.MILLISECONDS);
+        final List<Measurement> measurement = Collections.singletonList(new Measurement(() -> 2d, Statistic.VALUE));
+        Meter.builder(METER_NAME, Type.OTHER, measurement).tag(TEST_TAG_KEY, TEST_TAG_VALUE).register(registry);
+        final List<MetricsLogger> metricsLoggerList = registry.metricsLoggers();
+        assertThat(metricsLoggerList.size(), equalTo(9));
+    }
+
+    @Test
+    void registryMetricsLoggerDropEmptyTag() {
+        registry.gauge("gauge", Tags.of("empty", ""), 1d);
+        final List<MetricsLogger> metricsLoggerList = registry.metricsLoggers();
+        assertThat(metricsLoggerList.size(), equalTo(1));
+        final MetricsLogger metricsLogger = metricsLoggerList.get(0);
+        final MetricsContext context = reflectivelyGetMetricsContext(metricsLogger);
+        assertThat(hasDimension(context, "empty"), is(false));
+    }
+
+    @Test
+    void registryPublish() {
+        final MetricsLogger testMetricsLogger1 = mock(MetricsLogger.class);
+        final MetricsLogger testMetricsLogger2 = mock(MetricsLogger.class);
+        when(registry.metricsLoggers()).thenReturn(Arrays.asList(testMetricsLogger1, testMetricsLogger2));
+        registry.publish();
+        verify(testMetricsLogger1, times(1)).flush();
+        verify(testMetricsLogger2, times(1)).flush();
+    }
+
+    private MetricsContext reflectivelyGetMetricsContext(final MetricsLogger metricsLogger) {
+        final Field field;
+        try {
+            field = metricsLogger.getClass().getDeclaredField("context");
+            field.setAccessible(true);
+            final MetricsContext context = (MetricsContext) field.get(metricsLogger);
+            field.setAccessible(false);
+            return context;
+        } catch (final NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean hasDimension(final MetricsContext context, final String key) {
+        final List<DimensionSet> dimensionSetList = context.getDimensions();
+        for (final DimensionSet dimensionSet:dimensionSetList) {
+            if (dimensionSet.getDimensionKeys().contains(key)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void assertDimensionEqual(final MetricsContext context, final String key, final String value) {
+        final List<DimensionSet> dimensionSetList = context.getDimensions();
+        assertThat(dimensionSetList.size(), equalTo(1));
+        final DimensionSet dimensionSet = dimensionSetList.get(0);
+        assertThat(dimensionSet.getDimensionKeys().size(), equalTo(1));
+        assertThat(dimensionSet.getDimensionValue(key), equalTo(value));
+    }
+
+    private boolean HasMetric(final MetricsContext context, final Id meterId, final String suffix) throws JsonProcessingException {
```

Comment:
Function name should start with lower-case.

Reply:
Great catch!